src/index.js

```diff
@@ -14,8 +14,19 @@ import canvasManager from './objects/canvasManager.js';
  */
 window.setup = function()
 {
-    canvasManager.createCanvas();
-    sceneManager.setCurrentScene(MAIN_MENU);
+	let promises = [];
```

Comment:
There seems to be a lot of noise in this PR, is this the heart of the changes you're making here?

Reply:
I'll look into the noise.